src/Ocelot/Configuration/QoSOptions.cs

```diff
@@ -32,37 +32,83 @@ public QoSOptions(
             TimeoutValue = timeoutValue;
         }
 
+        public QoSOptions(
+            int exceptionsAllowedBeforeBreaking,
+            int durationOfBreak,
+            double failureRatio,
+            int timeoutValue,
+            string key)
+        {
+            DurationOfBreak = durationOfBreak;
+            ExceptionsAllowedBeforeBreaking = exceptionsAllowedBeforeBreaking;
+            Key = key;
+            TimeoutValue = timeoutValue;
+            FailureRatio = failureRatio;
+        }
+
+        public QoSOptions(
+            int exceptionsAllowedBeforeBreaking,
+            int durationOfBreak,
+            double failureRatio,
+            int samplingDuration,
+            int timeoutValue,
+            string key)
+        {
+            DurationOfBreak = durationOfBreak;
+            ExceptionsAllowedBeforeBreaking = exceptionsAllowedBeforeBreaking;
+            Key = key;
+            TimeoutValue = timeoutValue;
+            FailureRatio = failureRatio;
+            SamplingDuration = samplingDuration;
+        }
+
         /// <summary>How long the circuit should stay open before resetting in milliseconds.</summary>
         /// <remarks>If using Polly version 8 or above, this value must be 500 (0.5 sec) or greater.</remarks>
         /// <value>An <see cref="int"/> value (milliseconds).</value>
-        public int DurationOfBreak { get; } = DefaultBreakDuration;
+        public int DurationOfBreak { get; }
+
         public const int LowBreakDuration = 500; // 0.5 seconds
         public const int DefaultBreakDuration = 5_000; // 5 seconds
 
         /// <summary>
         /// How many times a circuit can fail before being set to open.
         /// </summary>
         /// <remarks>
-        /// If using Polly version 8 or above, this value must be 2 or greater.
+        /// if set to 0, the circuit-breaker is deactivated
+        /// otherwise this value must be 2 or greater. 
         /// </remarks>
         /// <value>
         /// An <see cref="int"/> value (no of exceptions).
         /// </value>
         public int ExceptionsAllowedBeforeBreaking { get; }
 
-        public string Key { get; }
+        /// <summary>
+        /// The failure-success ratio that will cause the circuit to break/open. 
+        /// </summary>
+        /// <value>
+        /// An <see cref="double"/> ratio of exceptions/requests  (0.8 means 80% failed of all sampled executions).
+        /// </value>
+        public double FailureRatio { get; } = .8;
```

Comment:
Actually [FailureRatio](https://www.pollydocs.org/api/Polly.CircuitBreaker.CircuitBreakerStrategyOptions-1.html#Polly_CircuitBreaker_CircuitBreakerStrategyOptions_1_FailureRatio) has the default value is 0.1
So, I don't recommend to override library settings!